src/App/Fossa/LicenseScanner.hs

```diff
@@ -99,22 +119,80 @@ scanVendoredDep ::
   , Has (Lift IO) sig m
   , Has StickyLogger sig m
   , Has Exec sig m
+  , Has ReadFS sig m
   ) =>
-  -- ApiOpts ->
   Path Abs Dir ->
   VendoredDependency ->
   m (NonEmpty LicenseUnit)
 scanVendoredDep baseDir VendoredDependency{..} = context "Scanning vendored deps for license data" $ do
   logSticky $ "License Scanning '" <> vendoredName <> "' at '" <> vendoredPath <> "'"
-  vendoredDepDir <- case parseRelDir $ toString vendoredPath of
-    Left err -> fatalText ("Error constructing scan dir for vendored scan: " <> showT err)
-    Right val -> pure val
-  let cliScanDir = baseDir </> vendoredDepDir
+  scanPath <- resolvePath' $ toString vendoredPath
+  case scanPath of
+    SomeFile (Abs path) -> scanArchive $ ScannableArchive path
+    SomeFile (Rel path) -> scanArchive . ScannableArchive $ baseDir </> path
```

Comment:
This doesn't work when you run `fossa analyze` from another directory.

E.g. if you run it like this:

```
cabal run fossa -- analyze --experimental-native-license-scan --debug -e http://localhost:9578 ~/fossa/license-scan-dirs/krb5-1.18-compressed
```

You get error messages like this:

```
Error reading file ./krb5-1.18.2.tar.gz:
              ./krb5-1.18.2.tar.gz: getFileStatus: does not exist (No such file or directory)
```

Happy to demo if this is hard to recreate

Reply:
Relative file paths should be based on the directory containing `fossa-deps.yml`.

Reply:
Are we picking up that file from the current directory, or the base directory?

Reply:
It looks like we're finding the `fossa-deps.yml` in the scan directory, but looking for the `.tar.gz` file in the current working directory.

In ~/fossa/license-scan-dirs/krb5-1.18-compressed I have two files: `fossa-deps.yml` and `krb5-1.18.2.tar.gz`

`fossa-deps.yml` looks like this:

```yaml
vendored-dependencies:
  - name: krb5.tar.gz
    path: ./krb5-1.18.2.tar.gz
    version: 1.18.2.1
```

I've tried with and without the `./` in the `path`, but we get the problem either way.

```
# cabal run fossa -- analyze --experimental-native-license-scan --debug -e http://localhost:9578 ~/fossa/license-scan-dirs/krb5-1.18-compressed
Up to date                                                                  
[DEBUG] Loading configuration file from "/Users/scott/fossa/license-scan-dirs/krb5-1.18-compressed/"
[DEBUG] License Scanning 'krb5.tar.gz' at 'krb5-1.18.2.tar.gz'                                                                                          
[ERROR] ----------                                                          
  An issue occurred                                                                                                                                     
                                                                            
  >>> Relevant errors                                                       
                                                                            
    Error                                                                   
                                                                            
      No native license scans were successful                               
                                                                                                                                                        
      Traceback:                                                            
        - Converting fossa-deps to partial API payload                                                                                                  
        - fossa-deps
        - fossa-analyze

  >>> Possibly-related warnings

    Warning

      >>> Relevant errors

        Error

          Error reading file krb5-1.18.2.tar.gz:
              krb5-1.18.2.tar.gz: getFileStatus: does not exist (No such file or directory)

          Traceback:
            - Scanning vendored deps for license data
            - Processing vendored dependency
            - Converting fossa-deps to partial API payload
            - fossa-deps
            - fossa-analyze
```

Reply:
Oh, and I manually added the change from https://github.com/fossas/fossa-cli/pull/865 and set `    enable_cli_license_scan: true,` in my local version of Core

Reply:
I think the problem is occuring on line 129. That's just using a relative path, right?

```
scanPath <- resolvePath' $ toString vendoredPath
```

Reply:
Yep, that's it.

Reply:
Can you try running from the current dir for the rest of the tests?  That's an easy fix, but it'll take a while before I can push it.

Reply:
Fixed in 9e02916